### PR TITLE
SAVAS-feat: BackgroundColor and Dialog tests

### PR DESCRIPTION
**Pull request by savaspamuk**

### CODE REVIEW

File: Lesson39/savas/src/components/DialogTest.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
+import * as React from "react";
```
Comment (annasvst) [inline Lesson39/savas/src/components/DialogTest.js:1]: You only need 'useState' here but import is importing everything from react which is not efficient. Try to never import more than you actually need.

File: Lesson39/savas/src/components/DialogTest.js [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+import DialogTitle from "@mui/material/DialogTitle";
+import Dialog from "@mui/material/Dialog";
+
+export default function DialogTest() {
+  const [openDialog, handleDisplay] = React.useState(false);
```
Comment (annasvst) [inline Lesson39/savas/src/components/DialogTest.js:6]:
```suggestion
  const [isDialogOpen, setIsDialogOpen] = useState(false);
```

File: Lesson39/savas/src/components/DialogTest.js [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+  const handleClose = () => {
+    handleDisplay(false);
```
Comment (annasvst) [inline Lesson39/savas/src/components/DialogTest.js:9]:
```suggestion
    setIsDialogOpen(false);
```

File: Lesson39/savas/src/components/DialogTest.js [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+  };
+
+  const openDialogBox = () => {
+    handleDisplay(true);
```
Comment (annasvst) [inline Lesson39/savas/src/components/DialogTest.js:13]:
```suggestion
    setIsDialogOpen(true);
```

File: Lesson39/savas/src/components/__test__/BackgroundColorChanger.test.js [hunk 5]
```diff
@@ -0,0 +1,20 @@
+import React from "react";
+import { render, fireEvent } from "@testing-library/react";
+import BackgroundColorChanger from "../BackgroundColorChanger";
+
+test("Background color changes on button click", () => {
+  const { getByText, container } = render(<BackgroundColorChanger />);
+
+  const greenButton = getByText("Green");
```
Comment (annasvst) [inline Lesson39/savas/src/components/__test__/BackgroundColorChanger.test.js:8]: Avoid getting elements by text when testing, this is an unreliable approach. Instead, add data-testid attributes as we did in the lesson and use `screen.getByTestId('your-data-testid)`.

File: Lesson39/savas/src/components/__test__/DialogTest.test.js [hunk 6]
```diff
@@ -0,0 +1,14 @@
+import React from "react";
+import { render, fireEvent } from "@testing-library/react";
+import DialogTest from "../DialogTest";
+
+test("When a button is clicked, a dialog should show up on a screen", () => {
+  const { getByText, queryByText } = render(<DialogTest />);
+
+  expect(queryByText("Test Dialog")).toBeNull();
```
Comment (annasvst) [inline Lesson39/savas/src/components/__test__/DialogTest.test.js:8]: Same there, use `data-testid` attributes

File: Lesson39/savas/src/components/__test__/DialogTest.test.js [hunk 7]
```diff
@@ -0,0 +1,14 @@
+import React from "react";
+import { render, fireEvent } from "@testing-library/react";
+import DialogTest from "../DialogTest";
+
+test("When a button is clicked, a dialog should show up on a screen", () => {
```
Comment (annasvst) [inline Lesson39/savas/src/components/__test__/DialogTest.test.js:5]: Add another test to test that the dialog can be closed correctly

**Review by annasvst — CHANGES_REQUESTED** [review]

File: Lesson39/savas/src/components/DialogTest.js (hunk 1 above)
Comment (savaspamuk) [reply Lesson39/savas/src/components/DialogTest.js:1]: Actually, I get this from 'tutorialspoint' and use it, but I understood what you suggested me :)

File: Lesson39/savas/src/components/__test__/BackgroundColorChanger.test.js (hunk 5 above)
Comment (savaspamuk) [reply Lesson39/savas/src/components/__test__/BackgroundColorChanger.test.js:8]: Here, if I use screen.getByTestId, the test gives 1 fail. I couldn't understand why it is?

File: Lesson39/savas/src/components/__test__/DialogTest.test.js (hunk 7 above)
Comment (savaspamuk) [reply Lesson39/savas/src/components/__test__/DialogTest.test.js:5]: I couldn't do it similar method, but I searched it and found 'waitFor' event. And it works :)

File: Lesson39/savas/src/components/__test__/BackgroundColorChanger.test.js [hunk 8]
```diff
@@ -0,0 +1,20 @@
+import React from "react";
+import { render, fireEvent } from "@testing-library/react";
+import BackgroundColorChanger from "../BackgroundColorChanger";
+
+test("Background color changes on button click", () => {
+  const { getByTestId, container } = render(<BackgroundColorChanger />);
```
Comment (annasvst) [inline Lesson39/savas/src/components/__test__/BackgroundColorChanger.test.js:6]: I don't see a good reason to extract getByTestId and container from render method. I'd recommend following official guide, see example here: https://testing-library.com/docs/react-testing-library/example-intro#full-example

Comment (savaspamuk) [reply Lesson39/savas/src/components/__test__/BackgroundColorChanger.test.js:6]: Were these changes appropriate?

**Review by annasvst — APPROVED** [review]

**Pull request merged**